### PR TITLE
Add capacitance calculation function for parallel plate capacitor with support for multiple inputs

### PR DESCRIPTION
**Pull request by dnxjay**
This pull request introduces a Python function that calculates the capacitance of a parallel plate capacitor based on the area of the plates and the distance between them. It includes  function capacitance(area, distance) to compute capacitance, Constant Definition, Support for Single and Multiple Inputs, Testing, Lambda and Map for Efficient Batch Processing

### CODE REVIEW

**Comment by laserlab** [comment]
@jkblc please review asap

**Review by jkblc — CHANGES_REQUESTED** [review]
Please address the linting problems

File: capacitance_module/__pycache__/capacitance.cpython-312.pyc
Comment (laserlab) [inline capacitance_module/__pycache__/capacitance.cpython-312.pyc:1]: no binary files, please remove

File: capacitance_module/capacitance.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#ance.py (extended with lambda and map)
```
Comment (laserlab) [inline capacitance_module/capacitance.py:1]: what is ance.py?

File: capacitance_module/capacitance.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+# Constant for permittivity of free space
+epsilon_0 = 8.854e-12  # F/m
+
+def capacitance(area, distance):
+    """
+    Calculate the capacitance of a parallel plate capacitor.
+
+    Parameters:
+    area (float): Area of the plate in square meters.
```
Comment (laserlab) [inline capacitance_module/capacitance.py:11]: do not specific a data type in python in the do string if that is not actually checked or necessary

File: capacitance_module/capacitance.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    distance (float): Distance between the plates in meters.
+
+    Returns:
+    float: Capacitance in farads.
+    """
+    return epsilon_0 * (area / distance)
+
+# Test the function with example values
+if __name__ == "__main__":
```
Comment (laserlab) [inline capacitance_module/capacitance.py:20]: Testing should be done in the unit test not in a main loop

File: capacitance_module/resistance.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# resistance.py
+
+def resistance(length, area, resistivity):
+    """
+    Calculate the resistance of a wire.
+
+    Parameters:
+    length (float): Length of the wire in meters.
```
Comment (laserlab) [inline capacitance_module/resistance.py:8]: see above

File: capacitance_module/resistance.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    float: Resistance in ohms.
+    """
+    return resistivity * (length / area)
+
+def batch_resistances(lengths, areas, resistivities):
+    """
+    Calculate resistances for a list of lengths, areas, and resistivities using map and lambda.
+
+    Parameters:
+    lengths (list of float): List of wire lengths.
```
Comment (laserlab) [inline capacitance_module/resistance.py:22]: list is useful, but not float

File: capacitance_module/test_resistance.py
Comment (laserlab) [inline capacitance_module/test_resistance.py:1]: docstrings missing

File: capacitance_module/test_capacitance.py
Comment (laserlab) [inline capacitance_module/test_capacitance.py:1]: docstrings missing

**Review by laserlab — CHANGES_REQUESTED** [review]
The actual homework task of using your electrodynamic function inside a function is not implemented.

**Comment by laserlab** [comment]
> Please address the linting problems. Otherwise looks good

Please check my review comments for issues the linter doesn’t catch

File: capacitance_module/test_capacitance.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    A = 1.0
+    d = 0.01
+    expected = epsilon_0 * (A / d)
+    assert math.isclose(capacitance(A, d), expected), f"Expected {expected}, but got {capacitance(A, d)}"
+
+    # Test batch capacitances
+    areas = [1.0, 2.0, 0.5]
+    distances = [0.01, 0.02, 0.005]
+    expected_batch = [epsilon_0 * (A / d) for A, d in zip(areas, distances)]
+    assert batch_capacitances(areas, distances) == expected_batch, "Batch capacitance test failed"
```
Comment (laserlab) [inline capacitance_module/test_capacitance.py:22]: Linter complains about, but in general you should actually run your tests …

**Review by jkblc — CHANGES_REQUESTED** [review]
You still need to use your electrodynamic function inside a function

File: capacitance_module/capacitance.py (hunk 3 above)
Comment (dnxjay) [reply capacitance_module/capacitance.py]: can you explain this part with more detail? I believe some of my issues to having a functional code at the moment is due to this

File: capacitance_module/__pycache__/capacitance.cpython-312.pyc
Comment (jkblc) [inline capacitance_module/__pycache__/capacitance.cpython-312.pyc:1]: Please remove the binary file

**Review by jkblc — CHANGES_REQUESTED** [review]
Looks better. Please remove the binary file

File: capacitance_module/capacitance.py
Comment (jkblc) [inline capacitance_module/capacitance.py:1]: You still need to use your electrodynamics function inside another function. That is an important part of the homework

Comment (dnxjay) [reply capacitance_module/capacitance.py:1]: Im confused about that, the code does include functions that use principles from electrodynamics inside other functions. The functions calculate_capacitance, calculate_capacitance_with_dielectric, and calculate_parasitic_capacitance all rely on electrodynamic formulas involving permittivity, area and distance between plates to calculate capacitance

Comment (jkblc) [reply capacitance_module/capacitance.py:1]: Maybe @laserlab can answer this, but each "def" in your code is a function. We are not directly talking about math functions.

Comment (jkblc) [reply capacitance_module/capacitance.py:1]: We discussed it in class today while talking about not assigning lambda functions to a variable

**Comment by laserlab** [comment]
@jkblc Is correct no use of ED function as an argument in another function. Not merging.

**Comment by dnxjay** [comment]
since mine wasnt merged, i dont get any credit for the work ive done???

**Comment by laserlab** [comment]
you do get points for the work, just not the merge points

**Pull request closed without merging**